### PR TITLE
feat: treasury_space_ui_example

### LINKED ISSUE #719: 🛠️ [TASK] : Treasury Space MVE1 Mocked UI
### Summary

Mocked Treasury Space Screen

### Description

Development of Treasury Space screen 

- [Figma](https://www.figma.com/design/80aDAQgoD0v06DO2Sb76ok/Catalyst--Minimal-Valuable-Experiments-(MVE's)?node-id=5152-60122&m=dev)

Part of #726

### PR DESCRIPTION
**Pull request by damian-molinski**
# Description

First iteration of `Treasury Space`. Right know only in UI Kit. Still needs to be implemented as proper screen in `Voices`

## Related Issue(s)

Part of #719 

## Description of Changes

UI Example page iteration

## Screenshots

https://github.com/user-attachments/assets/d78c2571-d449-43ef-91a3-4c9d814f36f4

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/lib/widgets/menu/voices_node_menu.dart [hunk 1]
```diff
@@ -5,7 +5,7 @@ import 'package:collection/collection.dart';
 import 'package:equatable/equatable.dart';
 import 'package:flutter/material.dart';
 
-final class VoicesNodeMenuItem extends Equatable {
+base class VoicesNodeMenuItem extends Equatable {
   const VoicesNodeMenuItem({
```
Comment (dtscalac) [inline catalyst_voices/lib/widgets/menu/voices_node_menu.dart:9]: For this class make sure to put class fields above constructor, not below

File: catalyst_voices/lib/widgets/menu/voices_node_menu.dart [hunk 2]
```diff
@@ -18,10 +18,58 @@ final class VoicesNodeMenuItem extends Equatable {
...
+}
+
+final class VoicesNodeMenuController extends ValueNotifier<VoicesNodeMenuData> {
+  set selected(int? newValue) {
+    value = newValue != null
+        ? value.copyWith(selectedItemId: newValue)
+        : value.clearSelection();
+  }
+
+  set isExpanded(bool newValue) {
```
Comment (dtscalac) [inline catalyst_voices/lib/widgets/menu/voices_node_menu.dart:61]: The project is following dart style convention, not flutter style convention. Getter / setter should be under constructor. Fields should be above the constructor.

File: catalyst_voices/uikit_example/lib/examples/treasury_space/voices_treasury_space.dart [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import 'package:flutter/material.dart';
+import 'package:uikit_example/examples/treasury_space/campaign_builder_panel.dart';
+import 'package:uikit_example/examples/treasury_space/campaign_comments_panel.dart';
+import 'package:uikit_example/examples/treasury_space/campaign_details.dart';
+import 'package:uikit_example/examples/treasury_space/voices_treasury_drawer.dart';
+
+final class CampaignSetupStep extends VoicesNodeMenuItem {
+  final String desc;
+
+  String get name => label;
```
Comment (dtscalac) [inline catalyst_voices/uikit_example/lib/examples/treasury_space/voices_treasury_space.dart:13]: getter below constructor according to dart convention

File: catalyst_voices/lib/widgets/drawer/voices_drawer_space_chooser.dart [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    required this.currentSpace,
+    required this.onChanged,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return VoicesDrawerChooser<Space>(
+      items: Space.values,
+      selectedItem: currentSpace,
+      onSelected: (value) => onChanged(value),
```
Comment (digitalheartxs) [inline catalyst_voices/lib/widgets/drawer/voices_drawer_space_chooser.dart:22]: simpler `onSelected: onChanged` reads better?

**Review by digitalheartxs — DISMISSED** [review]
Nice one 🙌

File: catalyst_voices/lib/widgets/drawer/voices_drawer_space_chooser.dart (hunk 4 above)
Comment (dtscalac) [reply catalyst_voices/lib/widgets/drawer/voices_drawer_space_chooser.dart:22]: btw, can we just write

```
onSelected: onChanged
```
?

File: catalyst_voices/lib/widgets/menu/voices_node_menu.dart (hunk 1 above)
Comment (damian-molinski) [reply catalyst_voices/lib/widgets/menu/voices_node_menu.dart:9]: It should be `final`. I was trying other approach but i didn't like it after all.
Will come back to `final`.

**Review by dtscalac — APPROVED** [review]
lgtm

**Pull request merged**